### PR TITLE
add explanation for what -w does in init context

### PR DESCRIPTION
**Pull request by emilypi**
Addresses #8103

### CODE REVIEW

File: cabal-install/src/Distribution/Client/Setup.hs [hunk 1]
```diff
@@ -2230,11 +2230,9 @@ initOptions _ =
     (reqArg' "TOOL" (Flag . (:[]))
                     (fromFlagOrDefault []))
 
-    -- NB: this is a bit of a transitional hack and will likely be
-    -- removed again if `cabal init` is migrated to the v2-* command
-    -- framework
   , option "w" ["with-compiler"]
-    "give the path to a particular compiler"
+    "give the path to a particular compiler. For 'init', this flag is used \
```
Comment (Mikolaj) [inline cabal-install/src/Distribution/Client/Setup.hs:2234]: Does this multiline string notation work here?

File: cabal-install/src/Distribution/Client/Setup.hs [hunk 2]
```diff
@@ -2230,11 +2230,9 @@ initOptions _ =
...
+    \to set the base bound inference."
```
Comment (Mikolaj) [inline cabal-install/src/Distribution/Client/Setup.hs:2235]: Given that `init` is also for newbies, perhaps something like "the 'base' package lower bound inference"?

File: cabal-install/src/Distribution/Client/Setup.hs (hunk 1 above)
Comment (emilypi) [reply cabal-install/src/Distribution/Client/Setup.hs:2234]: It should it's GHC spec

Comment (Mikolaj) [reply cabal-install/src/Distribution/Client/Setup.hs:2234]: TIL

File: cabal-install/src/Distribution/Client/Setup.hs (hunk 2 above)
Comment (emilypi) [reply cabal-install/src/Distribution/Client/Setup.hs:2235]: sure that makes sense

**Review by Mikolaj — APPROVED** [review]
LGTM

**Review by jneira — APPROVED** [review]
nicely resolved, thanks!

**Pull request merged**